src/wallet/wallet.cpp

```diff
@@ -2038,22 +2004,20 @@ CAmount CWalletTx::GetAvailableCredit(interfaces::Chain::Lock& locked_chain, boo
         }
     }
 
-    if (cache) {
-        *cache = nCredit;
-        assert(cache_used);
-        *cache_used = true;
+    if (fUseCache) {
```

Comment:
The old code appears to always store into the cache, as long as `filter` is `ISMINE_SPENDABLE` or `ISMINE_WATCHONLY`, regardless of the value of `fUseCache`.

Reply:
Oh, you're right. Good catch.